### PR TITLE
SUGCON24 EU: Fix Agenda and Sessions addition error bug

### PR DESCRIPTION
**Pull request by gkLeo**
## Description / Motivation

When adding the `Agenda` or `Sessions` in the Experience Editor, an error will occur. 

image

## How Has This Been Tested?

Local

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read the Contributing guide.
- [x] My code/comments/docs fully adhere to the Code of Conduct.
- [x] My change is a code change.
- [ ] My change is a documentation change and there are NO other updates required.

### CODE REVIEW

File: src/Project/Sugcon2024/Sugcon/src/components/Events/Agenda.tsx [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
 import useSWR from 'swr';
+import { ComponentProps } from 'lib/component-props';
 
 interface Fields {
   SessionizeUrl: Field<string>;
 }
 
-type AgendaProps = {
-  params: { [key: string]: string };
+type AgendaProps = ComponentProps & {
```
Comment (robearlam) [inline src/Project/Sugcon2024/Sugcon/src/components/Events/Agenda.tsx:11]: Adding the ComponentProps here, introduces errors in the related StoryBook assets, can these be updated as well?

File: src/Project/Sugcon2024/Sugcon/src/components/Events/Sessions.tsx [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
 import { Flex } from '@chakra-ui/react';
+import { ComponentProps } from 'lib/component-props';
 
 interface Fields {
   SessionizeUrl: Field<string>;
 }
 
-type SessionsProps = {
-  params: { [key: string]: string };
+type SessionsProps = ComponentProps & {
```
Comment (robearlam) [inline src/Project/Sugcon2024/Sugcon/src/components/Events/Sessions.tsx:12]: Adding the ComponentProps here, introduces errors in the related StoryBook assets, can these be updated as well?

File: src/Project/Sugcon2024/items/Renderings/Sugcon2024/Events/Agenda.yml [hunk 3]
```diff
@@ -18,13 +18,13 @@ SharedFields:
...
   Hint: __Read Only
-  Value: 0
+  Value: 
 - ID: "a77e8568-1ab3-44f1-a664-b7c37ec7810d"
   Hint: Parameters Template
   Value: "{1DFA17C5-0B40-4D43-A718-720BD65F65E6}"
 - ID: "b5b27af1-25ef-405c-87ce-369b3a004016"
   Hint: Datasource Location
-  Value: "query:./Data"
+  Value: "query:$site/*[@@name='Data']"
```
Comment (robearlam) [inline src/Project/Sugcon2024/items/Renderings/Sugcon2024/Events/Agenda.yml:27]: Why has this been moved to Site level?

File: src/Project/Sugcon2024/items/Renderings/Sugcon2024/Events/Sessions.yml [hunk 4]
```diff
@@ -18,13 +18,13 @@ SharedFields:
...
   Hint: __Read Only
-  Value: 0
+  Value: 
 - ID: "a77e8568-1ab3-44f1-a664-b7c37ec7810d"
   Hint: Parameters Template
   Value: "{C8DF4EA0-57FE-4D1C-B3A0-3E908FC6B34F}"
 - ID: "b5b27af1-25ef-405c-87ce-369b3a004016"
   Hint: Datasource Location
-  Value: "query:./Data"
+  Value: "query:$site/*[@@name='Data']"
```
Comment (robearlam) [inline src/Project/Sugcon2024/items/Renderings/Sugcon2024/Events/Sessions.yml:27]: Why has this been moved to Site level?

**Review by robearlam — CHANGES_REQUESTED** [review]

File: src/Project/Sugcon2024/Sugcon/src/components/Events/Agenda.tsx (hunk 1 above)
Comment (gkLeo) [reply src/Project/Sugcon2024/Sugcon/src/components/Events/Agenda.tsx:11]: Updated.

File: src/Project/Sugcon2024/Sugcon/src/components/Events/Sessions.tsx (hunk 2 above)
Comment (gkLeo) [reply src/Project/Sugcon2024/Sugcon/src/components/Events/Sessions.tsx:12]: Updated.

File: src/Project/Sugcon2024/items/Renderings/Sugcon2024/Events/Agenda.yml (hunk 3 above)
Comment (gkLeo) [reply src/Project/Sugcon2024/items/Renderings/Sugcon2024/Events/Agenda.yml]: Reverted.

File: src/Project/Sugcon2024/items/Renderings/Sugcon2024/Events/Sessions.yml (hunk 4 above)
Comment (gkLeo) [reply src/Project/Sugcon2024/items/Renderings/Sugcon2024/Events/Sessions.yml]: Reverted.

**Comment by gkLeo** [comment]
@robearlam StoryBook is updated.

**Review by robearlam — APPROVED** [review]

**Pull request merged**